paperoni/commands/command_collect_semantic_scholar.py

```diff
@@ -0,0 +1,281 @@
+        db = self.db
+        author_indices = []
+        topic_indices = []
+        # Paper: check MAG ID then title+abstract.
+        paper_id = (
+            db.select_id_from_values(
+                "paper_link",
+                "paper_id",
+                link_type="MAG",
+                link=paper.get_ref("MAG"),
+            )
+            or db.select_id_from_values(
+                "paper", "paper_id", title=paper.title, abstract=paper.abstract
+            )
+            or db.insert(
+                "paper",
+                ("title", "abstract", "citation_count"),
+                (paper.title, paper.abstract, paper.citation_count),
+            )
+        )
+        # Links
+        for link in paper.links:
+            if not db.count(
+                "paper_link",
+                "paper_id",
+                "link_type = ? AND link = ?",
+                (link.type, link.ref),
+            ):
+                db.insert(
+                    "paper_link",
+                    ("paper_id", "link_type", "link"),
+                    (paper_id, link.type, link.ref),
+                )
+        # Authors
+        for author in paper.authors:
+            # Author
+            author_id = db.select_id(
+                "author", "author_id", "author_name = ?", [author.name]
+            ) or db.insert("author", ["author_name"], [author.name])
+            author_indices.append((author_id, author))
+            # author links
+            for author_link in author.links:
+                db.modify(
+                    "INSERT OR IGNORE INTO "
+                    "author_link (author_id, link_type, link) VALUES(?, ?, ?)",
+                    (author_id, author_link.type, author_link.ref),
+                )
+            # Author affiliation.
+            for affiliation in author.affiliations:
+                # We don't have start and end date, so we check if
+                # affiliation and role are already registered with null dates.
+                if author.role is None:
+                    count = db.count(
+                        "author_affiliation",
+                        "author_id",
+                        "affiliation = ? AND role IS NULL "
+                        "AND start_date IS NULL and end_date IS NULL",
+                        [affiliation],
+                    )
+                else:
+                    count = db.count(
+                        "author_affiliation",
+                        "author_id",
+                        "affiliation = ? AND role = ? "
+                        "AND start_date IS NULL and end_date IS NULL",
+                        (affiliation, author.role),
+                    )
+                if not count:
+                    db.insert(
+                        "author_affiliation",
+                        ("author_id", "affiliation", "role"),
+                        (author_id, affiliation, author.role),
+                    )
```

Comment:
This part is not necessary, the author_affiliation table should be filled in a different way using internal information.